src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs

```diff
@@ -1478,6 +1456,84 @@ private static bool GetIsDaylightSavingsFromUtc(DateTime time, int year, TimeSpa
             return isDst;
         }
 
+        // This method checks if a specific year start with DST, if so, will return when the DST ends that year.
+        // Otherwise will return a default DateTime value.
+        private static DateTime GetEndOfDstIfYearStartWithDst(int nextYear, TimeSpan utc, TimeZoneInfo zone)
```

Comment:
(nit) what do you think about returning a `bool` and an `out DateTime` instead of using `default` to have a special value?

Reply:
I originally inplmented it this way and then decided to change it. The reason is I had to initialized the out param and then return the Boolean value. I mean the way I wrote it is a little bit less code. If you feel strong about it, I can change it back. I am not seeing big difference to object that :-)

Reply:
I don't have a strong opinion. I just dislike using special/sentinel values when I wouldn't have to.

Reply:
I applied your suggestion.